core/controller/src/main/scala/org/apache/openwhisk/core/controller/Triggers.scala

```diff
@@ -148,28 +148,32 @@ trait WhiskTriggersApi extends WhiskCollectionAPI {
             if (activeRules.nonEmpty) {
               val triggerActivationId = activationIdFactory.make()
               logging.info(this, s"[POST] trigger activation id: ${triggerActivationId}")
-              val triggerActivation = WhiskActivation(
-                namespace = user.namespace.name.toPath, // all activations should end up in the one space regardless trigger.namespace
-                entityName.name,
-                user.subject,
-                triggerActivationId,
-                Instant.now(Clock.systemUTC()),
-                Instant.EPOCH,
-                response = ActivationResponse.success(payload orElse Some(JsObject.empty)),
-                version = trigger.version,
-                duration = None)
-              val args: JsObject = trigger.parameters.merge(payload).getOrElse(JsObject.empty)
-
-              activateRules(user, args, trigger.rules.getOrElse(Map.empty))
-                .map(results => triggerActivation.withLogs(ActivationLogs(results.map(_.toJson.compactPrint).toVector)))
-                .recover {
-                  case e =>
-                    logging.error(this, s"Failed to write action activation results to trigger activation: $e")
-                    triggerActivation
-                }
-                .map { activation =>
-                  activationStore.store(activation, context)
-                }
+              if (user.limits.storeActivations.getOrElse(true)) {
```

Comment:
The `true` default is scattered throughout the codebase now. I'm wondering if there is a good way of centralizing that.

One way is to add a `val/def` to the `UserLimits` class, which defaults to true always.

The other way is to add a system "limit" which makes it possible to enable/disable this for the whole system, therefore acting as the default.

I realize though that the second approach might be a bigger undertaking. Maybe it's fine as-is and we're better off having the default at the actual caller side to later get rid off.

More generally speaking: We should think about building a more generalized cascading system for our limits. I envision something like this:

1. Load the user-limits from the database
2. Merge them with the system limits
3. Pass the merged versions through the code (maybe as part of the identity class)

That'd help us in cases as those and potentially transforms all the Option to actual values before we hit the location of usage.

You can ignore this comment if you want to, just wanted to leave the thoughts here.